presto-hive/src/test/java/com/facebook/presto/hive/TestHiveSkipEmptyFiles.java

```diff
@@ -0,0 +1,360 @@
+        Path partitionDirectory = Paths.get(tempDirectory + "/field=field1");
+        String partitionName = "field";
+        @Language("SQL") String insertQuery;
+        for (int i = 1; i <= 5; i++) {
+            insertQuery = format("INSERT INTO %s.\"%s\".\"%s\" VALUES (%s,'%s')",
+                    CATALOG, SCHEMA, tableName, i, partitionName + i);
+            queryRunner.execute(insertQuery);
+        }
+        if (replaceDataFileByEmptyFile) {
+            Files.delete(Arrays.stream(requireNonNull(partitionDirectory.toFile().listFiles((dir, name) -> !name.endsWith(".crc")))).iterator().next().toPath());
+        }
+        createTempFile(partitionDirectory, randomUUID().toString(), randomUUID().toString());
+    }
+
+    @Test
+    public void testSkipEmptyFilesSuccessful()
+    {
+        String tableName = "skip_empty_files_success";
+        executeCreationTestAndDropCycle(queryRunner, tableName, temporaryDirectory.toURI() + tableName);
+    }
+
+    @Test
+    public void testSkipEmptyFilesError()
+    {
+        String tableName = "skip_empty_files_fail";
+        executeCreationTestAndDropCycleFail(queryFailRunner, tableName, temporaryDirectory.toURI() + tableName);
+    }
+
+    @Test
+    public void testSkipEmptyFilesBucketSuccessful()
+    {
+        String tableName = "skip_empty_files_bucket_success";
+        checkBucketedResult(queryBucketRunner, tableName);
+    }
+
+    @Test
+    public void testSkipEmptyFilesBucketInsertFileFail()
+    {
+        String tableName = "skip_empty_files_bucket_insert_fail";
+        checkBucketedResultFail(queryBucketFailRunner, tableName, ".* is corrupt.* does not match the standard naming pattern, and the number of files in the directory .* does not match the declared bucket count.*");
+    }
+
+    @Test
+    public void testSkipEmptyFilesBucketReplaceFileFail()
+    {
+        String tableName = "skip_empty_files_bucket_replace_fail";
+        checkBucketedResultFail(queryBucketFailRunner, tableName, ".* is not a valid Parquet File");
+    }
+
+    /**
+     * Tries a table with the configuration property desired. If succeeds, tests the output.
+     * Finally, it drops the table.
+     *
+     * @param queryRunner a {@link QueryRunner} with the desired configuration properties
+     * @param tableName a {@link String} containing the desired table name
+     */
+    private void checkBucketedResult(DistributedQueryRunner queryRunner, String tableName)
+    {
+        try {
+            @Language("SQL") String selectQuery = format("SELECT * FROM %s.\"%s\".\"%s\"", CATALOG,
+                    SCHEMA, tableName);
+            MaterializedResult result = queryRunner.execute(selectQuery);
+            assertEquals(5, result.getRowCount());
+        }
+        finally {
+            @Language("SQL") String dropQuery = format("DROP TABLE IF EXISTS %s.\"%s\".\"%s\"", CATALOG,
+                    SCHEMA, tableName);
+            queryRunner.execute(dropQuery);
+        }
+    }
```

Comment:
Can we refactor this method and the one below to just check the output?  We can move the try/finally to the test cases themselves, and have a method in the finally clause that deletes the table.

Reply:
Also, did you also consider querying the `$files` hidden column?

Reply:
In this case it follows the same flow as a normal query and it doesn't add anything new.